File: integrations/aws-strands/error-code-corpus/typescript/code_through_reassignment.ts
/** A code the site reaches only through a later reassignment. */
import { EventType, type BaseEvent } from "@ag-ui/core";

function deriveCode(kind: string): string {
  return kind.toUpperCase();
}

export function emit(kind: string): BaseEvent {
  let failureCode = deriveCode(kind);
  failureCode = "THROUGH_REASSIGNMENT";
  return {
    type: EventType.RUN_ERROR,
    message: "Resolved through a reassignment.",
    code: failureCode,
  };
}

File: integrations/aws-strands/error-code-corpus/python/code_through_reassignment.py
"""A code the site reaches only through a later reassignment."""

from ag_ui.core import EventType, RunErrorEvent


def derive_code(kind: str) -> str:
    return kind.upper()


def emit(kind: str) -> RunErrorEvent:
    failure_code = derive_code(kind)
    failure_code = "THROUGH_REASSIGNMENT"
    return RunErrorEvent(
        type=EventType.RUN_ERROR,
        message="Resolved through a reassignment.",
        code=failure_code,
    )
